<a href="https://colab.research.google.com/github/aka1Neto/UEA/blob/main/Projeto_de_Matem%C3%A1tica.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import folium as fl;
import math;
from geopy.geocoders import Nominatim;
from geopy import distance;

In [ ]:
def locais(nome):
  geolocator = Nominatim(user_agent="Projeto");
  location = geolocator.geocode(nome);
  return location;

#Localizando os lugares desejados

In [ ]:
estado = locais(input("Digite o estado:\n"));
cidade1 = locais(input("Digite a primeira cidade:\n"));
cidade2 = locais(input("Digite a segunda cidade:\n"));
cidade3 = locais(input("Digite a terceira cidade:\n"));

Digite o estado:
amazonas
Digite a primeira cidade:
manaus
Digite a segunda cidade:
presidente figueiredo
Digite a terceira cidade:
itacoatiara


In [ ]:
def nome_cidade(cidade):
  lista_cidade = cidade.address.split(',');
  nome_cidade = lista_cidade[0];
  return nome_cidade;

In [ ]:
ncidade1 = nome_cidade(cidade1);
ncidade2 = nome_cidade(cidade2);
ncidade3 = nome_cidade(cidade3);

In [ ]:
def fazer_mapa():
  mapa = fl.Map(location=[estado.latitude,estado.longitude], zoom_start=6);
  fl.Marker(
      [cidade1.latitude, cidade1.longitude], popup=f'Cidade 1, {ncidade1}').add_to(mapa)
  fl.Marker(
      [cidade2.latitude, cidade2.longitude], popup=f'Cidade 2, {ncidade2}').add_to(mapa)
  fl.Marker(
      [cidade3.latitude, cidade3.longitude], popup=f'Cidade 3, {ncidade3}').add_to(mapa)
  return mapa;

In [ ]:
mapa = fazer_mapa();
mapa

# Distâncias

In [ ]:
def calcular_distancia(cid1, cid2):
  dist = distance.distance([cid1.latitude, cid1.longitude],[cid2.latitude, cid2.longitude]).km;
  return dist;

Distância entre cidade 1 e 2

In [ ]:
def fazer_rota(cid1, cid2, dist):
    map = fazer_mapa();
    fl.PolyLine([[cid1.latitude, cid1.longitude],[cid2.latitude, cid2.longitude]], popup=f'Distância: {dist:.2f}Km').add_to(map);
    return map;

In [ ]:
dist1 = calcular_distancia(cidade1, cidade2);
print(f'Distância entre {ncidade1} e {ncidade2} é de {dist1:.2f}Km\n')

Distância entre Manaus e Presidente Figueiredo é de 119.84Km



In [ ]:
mapa1 = fazer_rota(cidade1, cidade2, dist1);
mapa1

Distância entre cidade 2 e 3

In [ ]:
dist2 = calcular_distancia(cidade2, cidade3);
print(f'Distância entre {ncidade2} e {ncidade3} é de {dist2:.2f}Km\n')

Distância entre Presidente Figueiredo e Itacoatiara é de 213.42Km



In [ ]:
mapa2 = fazer_rota(cidade2, cidade3, dist2);
mapa2

Distância entre cidade 1 e 3

In [ ]:
dist3 = calcular_distancia(cidade1, cidade3);
print(f'Distância entre {ncidade1} e {ncidade3} é de {dist3:.2f}Km\n')

Distância entre Manaus e Itacoatiara é de 170.79Km



In [ ]:
mapa3 = fazer_rota(cidade1, cidade3, dist3);
mapa3

Triangulação da área

In [ ]:
coordenada = [
  (cidade1.latitude, cidade1.longitude),
  (cidade2.latitude, cidade2.longitude),
  (cidade3.latitude, cidade3.longitude),
  (cidade1.latitude, cidade1.longitude)
]
mapa4 = fazer_mapa();
fl.PolyLine(coordenada).add_to(mapa4);
mapa4

# Cálculos

Percebe-se então a formação de um triângulo retângulo, logo:

a² = b² + c² ou h² = co² + ca²

distPresItac² = distManaPres² + distManaItac²

distPresItac² = 120² + 171²

disPresItac² = 14400 + 29241

disPresItac² = 43641

disPresItac² = sqrt(43641)

disPresItac = 3sqrt(4849)

disPresItac é de aproximadamente 209Km

In [ ]:
x = math.sqrt((dist1**2) + (dist3**2));
print(f'A distância entre {ncidade2} e {ncidade3} é de {x:.2f}Km\n');

A distância entre Presidente Figueiredo e Itacoatiara é de 208.64Km



In [ ]:
distancias = [dist1, dist2, dist3];
valor_max = max(distancias);
valor_min = min(distancias);
print(f'O valor máximo é: {valor_max:.2f} e o valor mínimo é: {valor_min:.2f}\n')

O valor máximo é: 213.42 e o valor mínimo é: 119.84



In [ ]:
taxa_fixa = float(input("Qual o valor da taxa fixa?\n"));
taxa = float(input("Qual o valor da taxa?\n"));


Qual o valor da taxa fixa?
20
Qual o valor da taxa?
1.50


Trajeto de menor custo

Tmin = 20 + (120*1,5)

Tmin = 20 + 180

Tmin = 200

O trajeto de menor custo custa R$200,00

In [ ]:
tmin = taxa_fixa + (taxa*valor_min);
print(f'O menor trajeto custa {tmin:.2f}');

O menor trajeto custa 199.76


Trajeto de maior custo

Tmax = 20 + (214*1,5)

Tmin = 20 + 321

Tmin = 341

O trajeto de maior custo custa R$341,00

In [ ]:
tmax = taxa_fixa + (taxa*valor_max);
print(f'O maior trajeto custa {tmax:.2f}');

O maior trajeto custa 340.13


# Alternativas

A alternativa para a diminuição de custos, tendo somente a variação da distância, seria a troca de um trajeto em linha reta para um trajeto em curva, pois no globo, a menor distância entre dois pontos é uma curva.